cmd/panfs_config_cache.go

```diff
@@ -0,0 +1,90 @@
+// TODO: Add license header
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/minio/minio/internal/util/cache"
+)
+
+var errNotFoundInCache = errors.New("Item not found in cache")
+
+// CachingConfigAccessor can be used with the config operations implemented in
+// config-common.go (readConfig(), deleteConfig(), saveConfig()) to cache the
+// most frequently accessed data.
+type CachingConfigAccessor struct {
+	cache       *cache.Cache
+	rawAccessor configAccessor
+}
+
+func newCachingConfigAccessor(raw configAccessor) *CachingConfigAccessor {
```

Comment:
I'm not sure that we need eviction policy for bucket metadata at all. We have to know about all buckets and when the list is populated - we can fit it in memory. We will not have too many of them. Instead, I'd make sure that we invalidate cached metadata each time when we receive signal from mcproxy (there is signal handler in our fork of MinIO that reloads IAM cache).